src/borg/cache.py

```diff
@@ -564,6 +564,9 @@ def create_master_idx(chunk_idx):
             # deallocates old hashindex, creates empty hashindex:
             chunk_idx.clear()
             cleanup_outdated(cached_ids - archive_ids)
+            # Explicitly set the initial hash table capacity to avoid performance issues
+            # due to hash table "resonance".
+            master_index_capacity = int(len(self.repository) / ChunkIndex.MAX_LOAD_FACTOR)
```

Comment:
Also; it would probably be nicer if the capacity given would actually be the usable capacity, i.e. if hashindex_init or HashIndex.__init__ already added the necessary load factor margin.